### PR TITLE
improve timeout algo

### PR DESCRIPTION
**Pull request by holyxiaoxin**
default timeout from http poison is 8s.

### CODE REVIEW

File: lib/nadia/api.ex [hunk 1]
```diff
@@ -5,7 +5,7 @@ defmodule Nadia.API do
 
   alias Nadia.Model.Error
 
-  @default_timeout 4
+  @default_timeout 8
```
Comment (zhyu) [inline lib/nadia/api.ex]: https://github.com/edgurgel/httpoison/blob/87aabe18be27bf25af2a4182b4754cd0a35aa1dd/lib/httpoison/base.ex#L127-L128

default value for recv_timeout is 5s, so it's (4 + 1) right now.

**Comment by holyxiaoxin** [comment]
Update to configurable recv_timeout.

Improve algo would be `options[:timeout]+@default_timeout` (5s)
-> this makes sense because recv_timeout offset would be the same as default(5s) instead of current 1s.

File: README.md [hunk 2]
```diff
@@ -22,6 +22,12 @@ config :nadia,
   token: "bot token"
 ```
 
+You can also add an optional recv_timeout in seconds (defaults to 5s).
+```elixir
+config :nadia,
+  opt_recv_timeout: 10
```
Comment (zhyu) [inline README.md]: IMO, recv_timeout is whether default or custom, not optional.
So I'd like to use `recv_timeout` as the config name.

File: lib/nadia/api.ex [hunk 3]
```diff
@@ -54,7 +55,7 @@ defmodule Nadia.API do
   * `file_field` - specify the key of file_field in `options` when sending files
   """
   def request(method, options \\ [], file_field \\ nil) do
-    timeout = ((options[:timeout] || @default_timeout) + 1) * 1000
+    timeout = ((options[:timeout] || 0) + (opt_recv_timeout || @default_timeout)) * 1000
```
Comment (zhyu) [inline lib/nadia/api.ex]: a little bit complicated…
I'd like to get the fallback value in `opt_recv_timeout` method.

And I also prefer `get(source, key, deafult_value)` to 
`|| default_value`.
So I'll write like this:
```elixir
defp recv_timeout, do: Application.get_env(:nadia, :recv_timeout, @default_timeout)

timeout = (Keyword.get(options, :timeout, 0) + recv_timeout) * 1000
```

**Comment by zhyu** [comment]
I think it's better to have a working example in `config.exs.example`.
Could you add it?

**Comment by holyxiaoxin** [comment]
Sure. (:

**Comment by holyxiaoxin** [comment]
Do you still want the readme to contain the config: recv_timeout?

**Comment by zhyu** [comment]
Yes, just leave it there~

**Comment by zhyu** [comment]
Great work! Thx!

**Pull request merged**